docs/spec_draft.md

```diff
@@ -2126,6 +2128,87 @@ More formally, for each element `x`: `real(x) = is_complex(x) ? x.real : x`.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.recv
+
+### Semantics
+
+Receives data to a partition from a [stablehlo.send](#stablehlosend)
+instruction, in another partition, that shares the same channel handle
+`channel_handle`.
+
+`stablehlo.recv`, paired with the corresponding [stablehlo.send](#stablehlosend)
+instruction, represents synchronous communication. However, in order to enable
+asynchronous data transfers, it is decomposed into two HLO instructions
+(`Recv` and `RecvDone`). Similarly, the paired
+[stablehlo.send](#stablehlosend) is internally decomposed into two HLO
+instructions (`Send` and `SendDone`).
+
+These 4 instructions, each associated with a particular channel, has the
+following semantics:
+
+  * **Recv**: Initiator of the channel by allocating the receive buffer and
+              notifying the sender that it is ready to receive data.
+  * **Send**: Waits for the Recv notification and starts the data transfer to
+              the receive buffer.
+  * **SendDone**: Waits until the Send operation is done, so that the sender
+                  buffer can be reused for other purposes.
+  * **RecvDone**: waits until the data transfer is done and the receive buffer
+                  is ready for use.
+
+The diagram shows the execution order of the instructions.
+
+<img align="center" src="spec_draft/happens-before.svg" />
```

Comment:
in the image description
a -> b : Represents happens-before relationship A->B
can change to
A -> B : Represents the relationship, A happens-before B.